### PR TITLE
Feature/input assembly utility

### PR DESCRIPTION
**Pull request by djkapner**
An external collaborator was having trouble assembling an input json correctly for a list of experiments. this PR:
* adds an on-premises install option
* adds an on-premises utility to create an input json from a list of experiment IDs
* adds a short README to explain the above.
* makes the input/output schema more explicit rather than just specifying `Dicts`

Added the ability to query by session_id, which will only work for sessions/experiments that are 1-to-1 mappable (i.e. not mesoscope).
```
$ python -m nway.onprem_utils.assembly --experiment_ids 1023909329,1024382348,1025646202,1024072693 --output_json tmp.json
INFO:OnPremInputAssembly:loaded 19 ROIs for experiment 1023909329
INFO:OnPremInputAssembly:loaded 20 ROIs for experiment 1024382348
INFO:OnPremInputAssembly:loaded 23 ROIs for experiment 1025646202
INFO:OnPremInputAssembly:loaded 21 ROIs for experiment 1024072693
INFO:OnPremInputAssembly:Wrote tmp.json

$ python -m nway.onprem_utils.assembly --session_ids 1023831246,1024020218,1024303593,1024443480 --output_json tmp.json
INFO:OnPremInputAssembly:loaded 19 ROIs for experiment 1023909329
INFO:OnPremInputAssembly:loaded 21 ROIs for experiment 1024072693
INFO:OnPremInputAssembly:loaded 20 ROIs for experiment 1024382348
INFO:OnPremInputAssembly:loaded 25 ROIs for experiment 1024538071
INFO:OnPremInputAssembly:Wrote tmp.json
```

### CODE REVIEW

**Comment by djkapner** [comment]
@jeromelecoq @mhm34
Have a look at [this README](https://github.com/AllenInstitute/ophys_nway_matching/blob/feature/input_assembly_utility/src/nway/onprem_utils/README.md) for instructions on how to use this branch for your use case.

I'll get some feedback from you on the utility of it here, and some developer input also before merging to main. It may contnue to change until the merge.

File: src/nway/onprem_utils/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+### on-prem usage
+
+* install as you normally would, but include the optional `ONPREM`:
+```
+pip install .[ONPREM]
+```
+or
+```
+pip install -e .[ONPREM]
```
Comment (njmei) [inline src/nway/onprem_utils/README.md:9]: Hmm... I'm not a fan of providing end users the option of installing in editable mode. It's a really useful dev feature, but can really complicate troubleshooting with end users.

Maybe just remove?

File: src/nway/onprem_utils/query_utils.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        db_credentials[key] = os.environ.get(env_key, defaults[key])
+        if db_credentials[key] is None:
+            raise CredentialsException(
+                    "no ENV variable found and no default provided "
+                    f"for {env_key}")
+
+    return db_credentials
+
+
+class DbConnection():
```
Comment (njmei) [inline src/nway/onprem_utils/query_utils.py:66]: I feel like I've seen this code repeated quite often, we're a bit short on time currently, but maybe create a ticket that documents all the repeated instances so we can consolidate them into a utility package/repository...

**Review by njmei — APPROVED** [review]
Overall looks fine!

File: src/nway/onprem_utils/README.md (hunk 1 above)
Comment (djkapner) [reply src/nway/onprem_utils/README.md]: done

File: src/nway/onprem_utils/query_utils.py (hunk 2 above)
Comment (djkapner) [reply src/nway/onprem_utils/query_utils.py:66]: yeah, it is all over the place, including a stand-alone: https://github.com/AllenInstitute/simple-lims-connection
but, that repo is not public. I suppose this `[ONPREM]` option could require that private install, and only local users could do it anyway,

**Comment by djkapner** [comment]
Working, as verified by Jerome running it himself and sending results on to Mitch.

**Pull request merged**